src/platform/packages/shared/kbn-esql-utils/src/utils/cascaded_documents_helpers.ts

```diff
@@ -0,0 +1,489 @@
+  mutate.generic.commands.append(cascadeOperationQuery.ast, dataSourceCommand);
+
+  const newCommands = Object.entries(columnInterpolationRecord).map(([key, value]) => {
+    return Builder.command({
+      name: 'where',
+      args: [
+        Builder.expression.func.binary('==', [
+          Builder.expression.column({
+            args: [Builder.identifier({ name: key })],
+          }),
+          Builder.expression.literal.string(value),
+        ]),
+      ],
+    });
+  });
+
+  newCommands.forEach((command) => {
+    mutate.generic.commands.append(cascadeOperationQuery.ast, command);
+  });
+
+  return {
+    esql: BasicPrettyPrinter.print(cascadeOperationQuery.ast),
+  };
+}
+
+/**
+ * Handles the stats command for a leaf operation that contains a categorize function by modifying the query and adding necessary commands.
+ */
+function handleStatsByCategorizeLeafOperation(
+  dataSourceCommand: ESQLCommand<'from'>,
+  categorizeCommand: StatsFieldSummary,
+  nodePathMap: Record<string, string>
+) {
+  // create new query which we will modify to contain the valid query for the cascade experience
+  const cascadeOperationQuery = EsqlQuery.fromSrc('');
+
+  // append data source to to new query
+  mutate.generic.commands.append(cascadeOperationQuery.ast, dataSourceCommand);
+
+  // build a where command with match expressions for the selected categorize function
+  const categorizeWhereCommand = Builder.command({
+    name: 'where',
+    args: (categorizeCommand.definition as ESQLFunction<'variadic-call', 'categorize'>).args
+      .map((arg) => {
+        const namedColumn = categorizeCommand.column.name;
+
+        const matchValue = nodePathMap[removeBackticks(namedColumn)];
+
+        if (!matchValue) {
+          return null;
+        }
+
+        return Builder.expression.func.call('match', [
+          Builder.identifier({ name: (arg as ESQLColumn).text }),
+          Builder.expression.literal.string(extractCategorizeTokens(matchValue).join(' ')),
+          Builder.expression.map({
+            entries: [
+              Builder.expression.entry(
+                'auto_generate_synonyms_phrase_query',
+                Builder.expression.literal.boolean(false)
+              ),
+              Builder.expression.entry('fuzziness', Builder.expression.literal.integer(0)),
+              Builder.expression.entry('operator', Builder.expression.literal.string('AND')),
+            ],
+          }),
+        ]);
+      })
+      .filter(Boolean) as ESQLAstItem[],
+  });
+
+  mutate.generic.commands.append(cascadeOperationQuery.ast, categorizeWhereCommand);
+
+  return {
+    esql: BasicPrettyPrinter.print(cascadeOperationQuery.ast),
+  };
+}
+
+/**
+ * Modifies the provided ESQL query to only include the specified columns in the stats by option.
+ */
+export function mutateQueryStatsGrouping(query: AggregateQuery, pick: string[]): AggregateQuery {
+  const EditorESQLQuery = EsqlQuery.fromSrc(query.esql);
```

Comment:
When the query is parsed, most of the errors are silently reported as `.errors` array. You might want to check it:

```ts
EditorESQLQuery.errors
```